{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyN94kDHz86DxlToupZv/Rih"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","execution_count":1,"metadata":{"id":"-PnFbh7pgQCk","executionInfo":{"status":"ok","timestamp":1686650148711,"user_tz":-330,"elapsed":11804,"user":{"displayName":"Prema Research","userId":"05949325907362139936"}}},"outputs":[],"source":[" #A dependency of the preprocessing for BERT inputs\n","!pip install -q -U \"tensorflow-text==2.11.*\""]},{"cell_type":"code","source":["pip install -q tf-models-official==2.11.0"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"z3XV5WFKJ38Z","executionInfo":{"status":"ok","timestamp":1686650235754,"user_tz":-330,"elapsed":1801,"user":{"displayName":"Prema Research","userId":"05949325907362139936"}},"outputId":"9ee2ac04-8f7c-4be4-ac5b-407e4ab86527"},"execution_count":3,"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[31mERROR: Could not find a version that satisfies the requirement opencv-python-headless==4.5.2.52 (from tf-models-official) (from versions: 3.4.10.37, 3.4.11.39, 3.4.11.41, 3.4.11.43, 3.4.11.45, 3.4.13.47, 3.4.15.55, 3.4.16.59, 3.4.17.61, 3.4.17.63, 3.4.18.65, 4.3.0.38, 4.4.0.40, 4.4.0.42, 4.4.0.44, 4.4.0.46, 4.5.1.48, 4.5.3.56, 4.5.4.58, 4.5.4.60, 4.5.5.62, 4.5.5.64, 4.6.0.66, 4.7.0.68, 4.7.0.72)\u001b[0m\u001b[31m\n","\u001b[0m\u001b[31mERROR: No matching distribution found for opencv-python-headless==4.5.2.52\u001b[0m\u001b[31m\n","\u001b[0m"]}]},{"cell_type":"code","source":["!pip install protobuf --upgrade"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"DzjO6u53LO1b","executionInfo":{"status":"ok","timestamp":1686650036018,"user_tz":-330,"elapsed":8018,"user":{"displayName":"Prema Research","userId":"05949325907362139936"}},"outputId":"cabe46ad-b99c-4352-d039-c99ac94fcdca"},"execution_count":2,"outputs":[{"output_type":"stream","name":"stdout","text":["Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Requirement already satisfied: protobuf in /usr/local/lib/python3.10/dist-packages (3.19.6)\n","Collecting protobuf\n","  Downloading protobuf-4.23.2-cp37-abi3-manylinux2014_x86_64.whl (304 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m304.5/304.5 kB\u001b[0m \u001b[31m9.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hInstalling collected packages: protobuf\n","  Attempting uninstall: protobuf\n","    Found existing installation: protobuf 3.19.6\n","    Uninstalling protobuf-3.19.6:\n","      Successfully uninstalled protobuf-3.19.6\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","tensorboard 2.11.2 requires protobuf<4,>=3.9.2, but you have protobuf 4.23.2 which is incompatible.\n","tensorflow 2.11.1 requires protobuf<3.20,>=3.9.2, but you have protobuf 4.23.2 which is incompatible.\u001b[0m\u001b[31m\n","\u001b[0mSuccessfully installed protobuf-4.23.2\n"]}]},{"cell_type":"code","source":["!pip install tensorflow-datasets==4.8.0\n","!pip install tensorflow-metadata==1.12.0"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":1000},"id":"C3Wf6awkLd65","executionInfo":{"status":"ok","timestamp":1686650107383,"user_tz":-330,"elapsed":19169,"user":{"displayName":"Prema Research","userId":"05949325907362139936"}},"outputId":"170d2422-7d9d-4c65-c34b-f8905bec6180"},"execution_count":1,"outputs":[{"output_type":"stream","name":"stdout","text":["Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting tensorflow-datasets==4.8.0\n","  Downloading tensorflow_datasets-4.8.0-py3-none-any.whl (5.2 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m5.2/5.2 MB\u001b[0m \u001b[31m56.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: absl-py in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (1.4.0)\n","Requirement already satisfied: click in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (8.1.3)\n","Collecting dill (from tensorflow-datasets==4.8.0)\n","  Downloading dill-0.3.6-py3-none-any.whl (110 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m110.5/110.5 kB\u001b[0m \u001b[31m11.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: dm-tree in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (0.1.8)\n","Requirement already satisfied: etils[epath] in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (1.2.0)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (1.22.4)\n","Requirement already satisfied: promise in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (2.3)\n","Requirement already satisfied: protobuf>=3.12.2 in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (4.23.2)\n","Requirement already satisfied: psutil in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (5.9.5)\n","Requirement already satisfied: requests>=2.19.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (2.27.1)\n","Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (1.16.0)\n","Requirement already satisfied: tensorflow-metadata in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (1.13.1)\n","Requirement already satisfied: termcolor in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (2.3.0)\n","Requirement already satisfied: toml in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (0.10.2)\n","Requirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (from tensorflow-datasets==4.8.0) (4.65.0)\n","Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests>=2.19.0->tensorflow-datasets==4.8.0) (1.26.15)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests>=2.19.0->tensorflow-datasets==4.8.0) (2022.12.7)\n","Requirement already satisfied: charset-normalizer~=2.0.0 in /usr/local/lib/python3.10/dist-packages (from requests>=2.19.0->tensorflow-datasets==4.8.0) (2.0.12)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests>=2.19.0->tensorflow-datasets==4.8.0) (3.4)\n","Requirement already satisfied: importlib_resources in /usr/local/lib/python3.10/dist-packages (from etils[epath]->tensorflow-datasets==4.8.0) (5.12.0)\n","Requirement already satisfied: typing_extensions in /usr/local/lib/python3.10/dist-packages (from etils[epath]->tensorflow-datasets==4.8.0) (4.5.0)\n","Requirement already satisfied: zipp in /usr/local/lib/python3.10/dist-packages (from etils[epath]->tensorflow-datasets==4.8.0) (3.15.0)\n","Requirement already satisfied: googleapis-common-protos<2,>=1.52.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow-metadata->tensorflow-datasets==4.8.0) (1.59.0)\n","Installing collected packages: dill, tensorflow-datasets\n","  Attempting uninstall: tensorflow-datasets\n","    Found existing installation: tensorflow-datasets 4.9.2\n","    Uninstalling tensorflow-datasets-4.9.2:\n","      Successfully uninstalled tensorflow-datasets-4.9.2\n","Successfully installed dill-0.3.6 tensorflow-datasets-4.8.0\n","Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting tensorflow-metadata==1.12.0\n","  Downloading tensorflow_metadata-1.12.0-py3-none-any.whl (52 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m52.3/52.3 kB\u001b[0m \u001b[31m4.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: absl-py<2.0.0,>=0.9 in /usr/local/lib/python3.10/dist-packages (from tensorflow-metadata==1.12.0) (1.4.0)\n","Requirement already satisfied: googleapis-common-protos<2,>=1.52.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow-metadata==1.12.0) (1.59.0)\n","Collecting protobuf<4,>=3.13 (from tensorflow-metadata==1.12.0)\n","  Downloading protobuf-3.20.3-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.1 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.1/1.1 MB\u001b[0m \u001b[31m27.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hInstalling collected packages: protobuf, tensorflow-metadata\n","  Attempting uninstall: protobuf\n","    Found existing installation: protobuf 4.23.2\n","    Uninstalling protobuf-4.23.2:\n","      Successfully uninstalled protobuf-4.23.2\n","  Attempting uninstall: tensorflow-metadata\n","    Found existing installation: tensorflow-metadata 1.13.1\n","    Uninstalling tensorflow-metadata-1.13.1:\n","      Successfully uninstalled tensorflow-metadata-1.13.1\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","tensorflow 2.11.1 requires protobuf<3.20,>=3.9.2, but you have protobuf 3.20.3 which is incompatible.\u001b[0m\u001b[31m\n","\u001b[0mSuccessfully installed protobuf-3.20.3 tensorflow-metadata-1.12.0\n"]},{"output_type":"display_data","data":{"application/vnd.colab-display-data+json":{"pip_warning":{"packages":["google"]}}},"metadata":{}}]}]}